livepeer/studio
# www/multistream: Fix react hooks and some final touches

**What does this pull request do? Explain your changes. (required)**
This is to fix a couple bugs that went through to production on the multistream launch.

**Specific updates (required)**
Fixed bugs:
 - Menus/dialogs closing when the stream is active, on every stream/health refetch
 - Delete dialog bug that steals the window focus after closed

Implemented:
 - Edit target dialog (not a bug, but implemented it too!)
 - Also tried to consider stream session information on the status badge, but we actually
 expose no reliable API for that as of now. Still committed&reverted the POC just for 
 historical reasons, may want to pick it up again in the future.

## -

- **How did you test each of these updates (required)**
`yarn dev` and do all the multistream target flows:
 - Create a target
 - Enabled/disable
 - Edit target
 - Delete target
 - Start and stop an actual stream in between those steps and ensure UI is still usable and
 updates (like to name, source, url, etc) happen as expected.

**Does this pull request close any open issues?**
Closes https://github.com/livepeer/livepeer-com/issues/585

**Screenshots (optional):**
<img width="1487" alt="Screen Shot 2021-09-01 at 10 39 14" src="https://user-images.githubusercontent.com/1613383/131681596-4783f242-ca67-40bc-bef1-cc03e69275e8.png">
(the bugs can only really be tested)

**Checklist:**
